hathor/p2p/protocol.py

```diff
@@ -181,8 +182,11 @@ def is_state(self, state_enum: PeerState) -> bool:
 
     def get_short_remote(self) -> str:
         """Get remote for logging."""
-        assert self.transport is not None
-        return format_address(self.transport.getPeer())
+        return format_address(self.remote_address)
```

Comment:
If we want to make sure these have the same values we could check that `self.remote_address` and `self.transport.getPeer()` have the same value right when the transport is attributed to the `protocol` instance. Apparently this will happen _after_ `Factory.buildProtocol` is called, which is called by a `Connector`, which calls `protocol.makeConnection` on the instance returned by the factory. In our case this method is being inherited from `LineReceiver` into `HathorLineReceiver`, we could do something like:

```python
class HathorLineReceiver(LineReceiver, HathorProtocol):
    # ...
    def makeConnection(self, transport):
        assert self.remote_address == transport.getPeer()  # injected verification
        super().makeConnection(transport)
```

or we can also add it to `HathorLineReceiver.connectionMade` which is called when the transport has already been set, in which case we'd just use `self.transport` for the comparison.

Reply:
Great! I did on the `HathorProtocol.on_connect()`. Would you review it again, please?